pkg/volume/plugins.go

```diff
@@ -649,20 +649,12 @@ func (pm *VolumePluginMgr) FindPluginBySpec(spec *Spec) (VolumePlugin, error) {
 
 	matchedPluginNames := []string{}
 	matches := []VolumePlugin{}
-	for k, v := range pm.plugins {
-		if v.CanSupport(spec) {
-			matchedPluginNames = append(matchedPluginNames, k)
-			matches = append(matches, v)
-		}
-	}
+
+	matchedPluginNames, matches = matchPlugins(pm.plugins, spec, matchedPluginNames, matches)
 
 	pm.refreshProbedPlugins()
-	for _, plugin := range pm.probedPlugins {
-		if plugin.CanSupport(spec) {
-			matchedPluginNames = append(matchedPluginNames, plugin.GetPluginName())
```

Comment:
Since https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/plugins.go#L748 has already used the PluginName as the key in map,  so in this loop we could directly the key. Therefore, probedPlugins has the same logic as the plugins.